Xamarin.Forms.Core/Markup/Style.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace Xamarin.Forms.Markup
+{
+	public class Style<T> where T : Element // Should be IStyleElement but can't use that because it's internal
```

Comment:
Should we just expose `IStyleElement` for this?

Reply:
After looking into the Forms `Xamarin.Forms.Style` implementation I don't think so.

Technically, `Style` can be applied to any `BindableObject`; `IStyleElement` is only used to propagate `BasedOnResource` changes, but it is optional.

So instead of making this more restrictive by limiting it to `IStyleElement`, we should go the other direction and allow it to be used on any `BindableObject` instead of only `Element`.

Make sense?

Reply:
Sounds good to me! Thanks!

Reply:
Done